### ISSUE TITLE
New lint: preferred case for hex digit literals

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust-clippy#7987]
### What it does

Warns if {upper,lower}case hex digits are used in a literal (i.e. forces a particular case). That is, it is stronger than the `mixed_case_hex_literals` lint.

This is a "preference lint pair", like the `{un,}separated_literal_suffix` and `inline_asm_x86_{att,intel}_syntax` ones.

### Categories (optional)

- Kind: style.

### Drawbacks

In general, it is arguable whether this sort of "formatting-like" lints like `{un,}separated_literal_suffix` or the current one should be in Clippy vs. rustfmt (e.g. rustfmt already has a `hex_literal_case` option).

One advantage of having them as lints is that projects may `allow` them locally in certain cases.

### Example

```rust
0xabc
```

Could be written as:

```rust
0xABC
```

or viceversa.
